### PR TITLE
HAAR-1098 Align external user search with dps user search

### PR DESCRIPTION
**Pull request by karenmillermoj**

### CODE REVIEW

File: backend/api/oauthApi.js [hunk 1]
```diff
@@ -31,7 +31,6 @@ const oauthApiFactory = (client, { clientId, clientSecret, url }) => {
   const get = (context, path) => client.get(context, path).then((response) => response.body)
   const put = (context, path, body) => client.put(context, path, body).then((response) => response.body)
   const post = (context, path, body) => client.post(context, path, body).then((response) => response.body)
-  const del = (context, path) => client.del(context, path).then((response) => response.body)
```
Comment (karenmillermoj) [inline backend/api/oauthApi.js:34]: no longer used

File: backend/controllers/searchExternalUsers.js [hunk 2]
```diff
@@ -14,7 +17,7 @@ const searchFactory = (
   searchTitle,
   allowDownload,
 ) => {
-  const index = async (req, res) => {
+  return async (req, res) => {
```
Comment (karenmillermoj) [inline backend/controllers/searchExternalUsers.js:20]: we no longer have separate index and results as we always do a search for users (this matches dps user search)

File: backend/controllers/searchExternalUsers.test.js [hunk 3]
```diff
@@ -72,69 +89,83 @@ describe('search factory', () => {
...
 
   describe('results', () => {
     it('should call external search results render', async () => {
       const req = {
+        ...standardReq,
         query: { user: 'joe' },
         flash: jest.fn(),
         get: jest.fn().mockReturnValue('localhost'),
         protocol: 'http',
-        originalUrl: '/results',
```
Comment (karenmillermoj) [inline backend/controllers/searchExternalUsers.test.js:96]: no longer have a separate url as all search filtering and results are on the same page

File: backend/utils/nunjucksSetup.js [hunk 4]
```diff
@@ -245,6 +245,58 @@ module.exports = (app, path) => {
     },
   )
 
+  njkEnv.addFilter('toExternalUserSearchFilter', (currentFilter, groups, roles, filterOptionsHtml) => {
+    const hrefBase = '/search-external-users?'
+    const usernameTags = getUsernameTags(currentFilter, hrefBase)
```
Comment (karenmillermoj) [inline backend/utils/nunjucksSetup.js:250]: The filtering is similar to the dps user filtering

File: backend/utils/nunjucksSetup.js [hunk 5]
```diff
@@ -349,6 +401,35 @@ function getCaseloadTags(currentFilter, hrefBase, prisons) {
 
   return tags
 }
+
+function getRoleCodeNameTags(currentFilter, hrefBase, roles) {
+  const { roleCode, ...newFilter } = currentFilter
+  if (roleCode) {
+    return [
+      {
+        // TODO look at using new URLSearchParams instead
```
Comment (karenmillermoj) [inline backend/utils/nunjucksSetup.js:410]: All other methods in here have the TODO

**Review by AlTurner-MOJ — APPROVED** [review]
lgtm

**Pull request merged**